### PR TITLE
enhance: ban range-search iteration for search-group-by

### PR DESCRIPTION
**Pull request by MrPresent-Han**
related: #30033

### CODE REVIEW

**Comment by MrPresent-Han** [comment]
rerun ut

**Comment by MrPresent-Han** [comment]
rerun ut

**Comment by MrPresent-Han** [comment]
rerun ut

**Comment by MrPresent-Han** [comment]
rerun ut

File: internal/proxy/task_search.go [hunk 1]
```diff
@@ -207,6 +207,17 @@ func parseSearchInfo(searchParamsPair []*commonpb.KeyValuePair, schema *schemapb
 		}
 	}
 
+	// 6. parse iterator tag, prevent trying to groupBy when doing iteration or doing range-search
+	isIterator, _ := funcutil.GetAttrByKeyFromRepeatedKV(IteratorField, searchParamsPair)
+	if isIterator == "True" && groupByFieldId > 0 {
```
Comment (congqixia) [inline internal/proxy/task_search.go:212]:
```suggestion
	if strings.ToLower(isIterator) == "true" && groupByFieldId > 0 {
```

**Review by congqixia — APPROVED** [review]
/lgtm
please fix the comment in next PR

**Pull request merged**